l10n_br_account/models/account_fiscal_position_rule.py

```diff
@@ -75,12 +101,26 @@ def _map_domain(self, partner, addrs, company, **kwargs):
         document_date = self.env.context.get('date', time.strftime('%Y-%m-%d'))
         use_domain = self.env.context.get(
             'use_domain', ('use_sale', '=', True))
-
-        domain = [
-            '&', ('company_id', '=', company.id), use_domain,
+        special_domain = []
+        domain = ['&']
+        for special in self.env[
+            'l10n_br_account.partner.special.fiscal.type'
+        ].search([]).ids:
+            if special in partner.partner_special_fiscal_type_id.ids:
+                special_domain.append(
+                    ('partner_special_fiscal_type_id', 'in', special))
+            else:
+                special_domain.append(
+                    ('partner_special_fiscal_type_id', 'not in', special))
+        if special_domain:
+            domain += special_domain
+        else:
+            domain += [('partner_special_fiscal_type_id', '=', False)]
+        domain += [
+            ('company_id', '=', company.id), use_domain,
             ('fiscal_type', '=', company.fiscal_type),
             ('fiscal_category_id', '=', kwargs.get('fiscal_category_id')),
-            '|', ('partner_fiscal_type_id', '=', partner_fiscal_type_id),
+            '|', ('partner_fiscal_type_id', 'in', partner_fiscal_type_id),
```

Comment:
Acredito que é melhor manter o operador '=' ao invés de troca-lo para 'in' já que o campo recebe um id ( ex.: 1 ) e não uma lista